forge/db/controllers/Device.js

```diff
@@ -12,29 +12,35 @@ module.exports = {
         return device.activeSnapshotId !== device.targetSnapshotId
     },
     updateState: async function (app, device, state) {
-        if (state.state) {
-            device.set('state', state.state)
-        }
-        if (state.agentVersion) {
-            device.set('agentVersion', state.agentVersion)
-        }
         device.set('lastSeenAt', literal('CURRENT_TIMESTAMP'))
-        if (!state.snapshot || state.snapshot === '0') {
-            if (device.activeSnapshotId !== null) {
-                device.set('activeSnapshotId', null)
-            }
+        if (!state) {
+            // We have received a `null` state from the device. That means it
+            // is busy updating itself. Update out local state to infer as much
+            // as we can from that
+            device.set('state', 'updating')
```

Comment:
```suggestion
        if (!state) {
            // We have received a `null` state from the device. That means it
            // is busy updating itself. Update our local state to infer as much
            // as we can from that
            device.set('state', 'updating')
```

Super minor typo (had to read comment several times)